### PR TITLE
Update Zenoh Spec with work flow

### PR DESCRIPTION
**Pull request by evshary**
As @PLeVasseur requested in #81, add the work flow in Zenoh Spec.

### CODE REVIEW

File: up-l1/zenoh.adoc [hunk 1]
```diff
@@ -178,3 +183,91 @@ Encoding::WithSuffix(
...
+1. A uEntity (`origin-authority/origin-entity`) creates a message with
+    * source: `up://origin-authority/origin-entity/origin-version/origin-resource.instance`
+    * sink: `up://dest-authority/dest-entity/dest-version/dest-resource.instance`
+
+2. `UTransport.send` maps sink to Zenoh key
+    * local (w/o dest-authority): `upl/` + `dest-resource-id + dest-entity-id + dest-version`
+    * remote (with dest-authority): `upr/` + `dest-authority-id/` + `dest-resource-id + dest-entity-id + dest-version`
+
+3. Receiver (dest-authority/dest-entity) registers a listener for
+    * `up://dest-authority/dest-entity/dest-version/dest-resource.instance`
```
Comment (stevenhartley) [inline up-l1/zenoh.adoc:221]: Receiver registers a listener for `up://origin-authority/origin-entity/origin-version/origin-resource.instance` . When the message type is UMESSAGE_TYPE_NOTIFICATION, the up-client library matches the source (not the sink) to the listner.

File: up-l1/zenoh.adoc [hunk 2]
```diff
@@ -178,3 +183,91 @@ Encoding::WithSuffix(
...
+
+4. which `UTransport.registerListener` maps to Zenoh key
```
Comment (stevenhartley) [inline up-l1/zenoh.adoc:223]: see comment above.

**Review by stevenhartley — CHANGES_REQUESTED** [review]
Thank you for this CY, I think this will drastically help explain and I need to put this high level explainations in the UTransport API specification if we agree. 

Publish and notifications listener matching is to the source (client side), but request and response are to the sink.

File: up-l1/zenoh.adoc (hunk 2 above)
Comment (stevenhartley) [reply up-l1/zenoh.adoc:223]: Michael identified a key issue now on the client side in registerListener() that I'll document in #81

File: up-l1/zenoh.adoc [hunk 3]
```diff
@@ -178,3 +183,91 @@ Encoding::WithSuffix(
...
+    * remote (with origin-authority): `upr/` + `origin-authority-id/` + `origin-resource-id + origin-entity-id + origin-version`
+
+3. Subscriber registers a listener for
+    * `up://origin-authority/origin-entity/origin-version/origin-resource.instance#message`
+
+4. which `UTransport.registerListener` maps to Zenoh key
+    * local (w/o origin-authority): `upl/` + `origin-resource-id + origin-entity-id + origin-version`
+    * remote (with origin-authority): `upr/` + `origin-authority-id/` + `origin-resource-id + origin-entity-id + origin-version`
+
+5. Since this is **Publish URI**, `Zenoh subscriber` will be used to bind the `registerListener`
```
Comment (PLeVasseur) [inline up-l1/zenoh.adoc:208]: For the section on `registerListener` for Publish, Notification, Request, and Response I think it'd be a good idea to explicitly make note of which UUri we're registering on: destination or origin.

**Comment by PLeVasseur** [comment]
@MelamudMichael -- could you please review this so that the cpp and rust implementations are on the same page?

File: up-l1/zenoh.adoc (hunk 3 above)
Comment (evshary) [reply up-l1/zenoh.adoc:208]: I agree with you. It's clearer to readers.

**Comment by PLeVasseur** [comment]
IMHO for traceability would be nice to keep separate, but I don't have a very strong opinion =^)

**Review by PLeVasseur — APPROVED** [review]
Yup, makes sense. Thanks for updating

**Comment by evshary** [comment]
@PLeVasseur No. As you suggested [here](https://github.com/eclipse-uprotocol/up-spec/pull/92#issuecomment-2025171681), I'm creating another PR for the changes.
It would be good if this PR could be merged first.

**Review by stevenhartley — APPROVED** [review]
#103 will be used to fix the other part of the spec w.r.t URI mapping to zenoh keys

**Pull request merged**